enterprise/security/src/test/java/org/neo4j/server/security/enterprise/auth/integration/bolt/LdapAuthenticationIT.java

```diff
@@ -513,6 +525,82 @@ public void shouldBeAbleToUseProcedureAllowedAnnotationWithLdapGroupToRoleMappin
         assertAllowedReadProcedure();
     }
 
+    @Test
+    public void shouldTimeoutIfInvalidLdapServer() throws Throwable
+    {
+        // When
+        restartNeo4jServerWithOverriddenSettings( ldapOnlyAuthSettings.andThen( settings -> {
+            settings.put( SecuritySettings.ldap_server, "ldap://198.51.100.9" ); // An IP in the TEST-NET-2 range
+            settings.put( SecuritySettings.ldap_connection_timeout, "1s" );
+        } ) );
+
+        assertConnectionTimeout( authToken( "neo", "abc123", null ),
+                LDAP_CONNECTION_TIMEOUT_CLIENT_MESSAGE );
+    }
+
+    @Test
+    @Category( TimeoutTests.class )
```

Comment:
You can categorise the tests to run only a subset. I wanted to run only the timeout tests when I was debugging.